atlp-rwanda/e-comm-team-amigos-bn
# #184797861 Place Order

#### What does this PR do?

This PR allow user create, update, and delete order

#### Description of Task to be completed?

Have the following endpoints working

- [x] `POST /orders`
- [x] `GET /orders`
- [x] `PATCH /orders/<order-id>`
- [x] `DELETE /orders/<order-id>`

#### How should this be manually tested?

- In your **e-comm-team-amigos-bn** directory, open terminal and run `git fetch origin pull/55/head:ft-order-for-buyers-18479786`
- Checkout to **ft-order-for-buyers-18479786** branch
- Run `npm install`
- Run `npm run dev`
- In your working directory terminal, run migrations and seeders. **NB:** Run seeder whose name contain demo order. But, first, make sure your **normal users** have a similar password.
- In **postman** do the following
     - Sigup or Sign in with customer account
     - Make a request on `POST /orders` with req.body like so `{items: [{product: <product-id>, quantity: <product-quantity>},  unitPrice: <price>]}`
     - Make a request on `PATCH /orders/<order-id>` passing new data
     - Make a request on `DELETE /orders/<order-id>`

#### Any background context you want to provide?
- Should have your `.env` file with neccessary variables as in `.env.example`
     
#### What are the relevant pivotal tracker/Trello stories?

Story ID: [184797861](https://www.pivotaltracker.com/story/show/184797861)

### Screenshots

`Tests`
![Screenshot (1498)](https://user-images.githubusercontent.com/55640083/228881804-9e532169-1fd6-4005-80bf-204935b3d967.png)

`Doc`
![Screenshot (1500)](https://user-images.githubusercontent.com/55640083/228882224-5343fa0a-d6ba-4eae-967d-29b962567607.png)
